Assets/Scripts/Calculator.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Text;
+
+// The calculator that executes input expressions.
+public class Calculator {
+  private const int _maxChars = 100;
+  private readonly CalculatorScreen _screen = null;
+  private readonly StringBuilder _cachedInput = new StringBuilder();
+
+  private bool _hasError = false;
+
+  public bool HasError => _hasError;
```

Comment:
If "public bool HasError { get; private set; } = false" simpler than these two lines of code?

Reply:
Done.